src/broadcast.rs

```diff
@@ -89,19 +86,68 @@ pub struct Broadcast<NodeUid> {
     /// Whether we have already output a value.
     has_output: bool,
     /// The proofs we have received via `Echo` messages, by sender ID.
-    echos: BTreeMap<NodeUid, Proof<Vec<u8>>>,
+    echos: BTreeMap<N, Proof<Vec<u8>>>,
     /// The root hashes we received via `Ready` messages, by sender ID.
-    readys: BTreeMap<NodeUid, Vec<u8>>,
+    readys: BTreeMap<N, Vec<u8>>,
+    /// The outgoing message queue.
+    messages: VecDeque<TargetedMessage<BroadcastMessage, N>>,
+    /// The output, if any.
+    output: Option<Vec<u8>>,
 }
 
-impl<NodeUid: Eq + Hash + Debug + Clone + Ord> Broadcast<NodeUid> {
+impl<N: Eq + Debug + Clone + Ord> DistAlgorithm for Broadcast<N> {
+    type NodeUid = N;
+    type Input = Vec<u8>; // TODO: Allow anything serializable.
```

Comment:
Can you comment a bit more on what you mean by "serializable". Is it `Serde::Serialize`?

Reply:
Yes, and deserializable, too; something like I already did in honey_badger.rs: `T: serde::Serialize + serde::de::DeserializeOwned`.

The downside of this is that it depends on serde unconditionally. I think that's okay, since serde is extremely widely used anyway, but alternatively we could make this also depend on a feature flag later, and e.g. optionally use `T: protobuf::Message` instead.